### PR TITLE
Port ArchivedPatient to pg

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Port over ArchivedPatient model and the associated goods.

To test, you can just do the migration stuff:

```
git checkout main && rails db:drop db:create db:migrate db:seed
git checkout pg-archived-pt && rails db:migrate migrate_to_pg:clinic_user_patient
```

Should yield the following:

```
4 Clinic migrated to pg
3 User migrated to pg
39 Patient migrated to pg
4 ArchivedPatient migrated to pg
39 Fulfillment migrated to pg
141 calls migrated to pg
0 practical_supports migrated to pg
2 external_pledges migrated to pg
43 Fulfillment migrated to pg
149 calls migrated to pg
0 practical_supports migrated to pg
2 external_pledges migrated to pg
6 CallListEntry migrated to pg
17 notes migrated to pg
```

After this, we're mostly in the domain of getting tests to work I think?

This pull request makes the following changes:
* Port ArchivedPatient to pg

no view changes

It relates to the following issue #s: 
* Bumps #2072 

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

**Comment by colinxfleming** [comment]
Updated yield:

```
4 Clinic migrated to pg
3 User migrated to pg
37 Patient migrated to pg
2 ArchivedPatient migrated to pg
37 Fulfillment migrated to pg
137 calls migrated to pg
0 practical_supports migrated to pg
2 external_pledges migrated to pg
39 Fulfillment migrated to pg
141 calls migrated to pg
0 practical_supports migrated to pg
2 external_pledges migrated to pg
6 CallListEntry migrated to pg
16 notes migrated to pg
```

**Comment by colinxfleming** [comment]
(Also updated commands)

```
```
git checkout main && rails db:drop db:create db:migrate db:seed

rails c
ArchivedPatient.destroy_all
ArchivedPatient.archive_eligible_patients!
ArchivedPatient.count #=> 2
exit

git checkout pg-archived-pt && rails db:migrate migrate_to_pg:clinic_user_patient
```

**Review by lomky — APPROVED** [review]
Approved pending the VM enum!

File: app/models/archived_patient.rb [hunk 1]
```diff
@@ -1,87 +1,50 @@
...
-  field :identifier, type: String # UUID, not phone number based!
-
-  # Fields generated from initial patient info
-  field :has_alt_contact, type: Mongoid::Boolean
-  field :age_range
-  enumerize :age_range, in: [:not_specified, :under_18, :age18_24, :age25_34, :age35_44, :age45_54, :age55plus, :bad_value], default: :not_specified
-
-  # Fields pulled from initial Patient
-  field :voicemail_preference
-  enumerize :voicemail_preference, in: [:not_specified, :no, :yes], default: :not_specified
```
[the remark this answers is not in the document]
Comment (colinxfleming) [reply app/models/archived_patient.rb:30]: The latter -- we lost it and are treating it like a string nowadays. Patient follows this pattern on main I think; ArchivedPatient is lagging a bit behind.

**Comment by colinxfleming** [comment]
thank you as always @lomky !

**Pull request merged**